### PR TITLE
REST Endpoint Phase 1

### PR DESCRIPTION
**Pull request by logotie**
Sourced from: #4  

On sending a Post request to '/torrentfile' with:

Header Token - 'Authorize' must have the text 'TEST'
with the json body of 
{"url":""}
You can put whatever is needed in the url field.

You will receive back
{"id":"####-#####"}
### signifying a GUID code.

-- I added some XUnit tests that are currently passing.  

Please let me know, if there are any improvements/changes you want made

### CODE REVIEW

**Comment by Earu** [comment]
List of required changes:
- `String.Equals(arg1, arg2)` is not idiomatic of the C# lang, use `arg1.Equals(arg2)` instead 
- Group your services and their interfaces into one interface and service class
- Remove `DbContext` as it not useful 
- Remove `WebConstant` and instead index the headers directly with either the relevant enum or the "Authorization" string

**Comment by logotie** [comment]
Thanks @Earu for taking the time out to review the code!

I will make these changes

**Comment by logotie** [comment]
Hiya @Earu, I have made the requested changes :)

File: Sulfur/Services/UrlHeaderActions/UrlActionService.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        //Checks whether the auth header token matches our predefined value
+        public bool SuccessfulMatchOnHeaderToken(string headerToken)
+        {
+            return headerToken.Equals(ServiceConstants.AuthTokenPassword);
+        }
+
+        //Create a new Guid code to be sent back as a response.
+        //TODO In the future, the UrlPayload will be 
+        GuidResult IUrlAactionService.GenerateGuidPayload()
```
Comment (Earu) [inline Sulfur/Services/UrlHeaderActions/UrlActionService.cs]: Whats the visibility of this method ? Public ? Annotate it please
Remove the IUrlAactionService

File: Sulfur/Services/UrlHeaderActions/UrlActionService.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿using Sulfur.Constants;
+using Sulfur.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Sulfur.Services.UrlHeaderActions
+{
+    public class UrlHeaderService : IUrlAactionService
```
Comment (Earu) [inline Sulfur/Services/UrlHeaderActions/UrlActionService.cs]: I think you made a typo in the same of the interface

File: Sulfur/Services/UrlHeaderActions/UrlActionService.cs (hunk 1 above)
Comment (logotie) [reply Sulfur/Services/UrlHeaderActions/UrlActionService.cs]: Ah apologies , will do

File: Sulfur/Services/UrlHeaderActions/UrlActionService.cs (hunk 2 above)
Comment (logotie) [reply Sulfur/Services/UrlHeaderActions/UrlActionService.cs]: Ah yeah, thanks!

**Comment by logotie** [comment]
Alright, I have made the changes :)

**Comment by logotie** [comment]
Thanks!  I will write up Phase 2 features

**Pull request merged**